### PR TITLE
Quarantine failing test

### PR DESCRIPTION
**Pull request by wtgodbe**
https://github.com/dotnet/aspnetcore/issues/25206

### CODE REVIEW

**Review by dougbu — APPROVED** [review]

**Comment by dougbu** [comment]
Do all variants of this `[Theory]` fail when one fails❔ I ask because we get a fair amount of coverage out of testing the very different samples used here.

**Comment by wtgodbe** [comment]
Only one variant failed, but I can't find info on which one it was: https://dev.azure.com/dnceng/public/_build/results?buildId=784906&view=ms.vss-test-web.build-test-results-tab&runId=24707436&resultId=101347&paneView=debug

**Comment by dougbu** [comment]
file:///C:/Users/dougbu/Downloads/Microsoft.AspNetCore.Mvc.FunctionalTests_net5.0_x64.html#failures shows it was only the `image` test that failed. Suggest splitting that into a `[Fact]` that calls the same base test method and quarantining the new method.

Also, since we expect some macOS slowness these days, suggest bumping the timeout duration up.

**Comment by dougbu** [comment]
Also suggest quarantining only on macOS

File: src/Mvc/test/Mvc.FunctionalTests/HtmlGenerationTest.cs [hunk 1]
```diff
@@ -140,6 +137,15 @@ public async Task HtmlGenerationWebSite_GeneratesExpectedResults(string action,
             }
         }
 
+        [Fact]
+        [InlineData("Image", null)]
+        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/25206")]
+        public async Task HtmlGenerationWebSite_GeneratesExpectedResults_WithImageData(string action, string antiforgeryPath)
+        {
+            HtmlGenerationWebSite_GeneratesExpectedResults(action, antiforgeryPath);
+        }
```
Comment (dougbu) [inline src/Mvc/test/Mvc.FunctionalTests/HtmlGenerationTest.cs]:
```suggestion
        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/25206")]
        public async Task HtmlGenerationWebSite_GeneratesExpectedResults_WithImageData() =>
            HtmlGenerationWebSite_GeneratesExpectedResults("image", antiforgeryPath: null);
```

File: src/Mvc/test/Mvc.FunctionalTests/HtmlGenerationTest.cs [hunk 2]
```diff
@@ -140,6 +137,15 @@ public async Task HtmlGenerationWebSite_GeneratesExpectedResults(string action,
...
+
+
```
Comment (dougbu) [inline src/Mvc/test/Mvc.FunctionalTests/HtmlGenerationTest.cs:148]: nit: Remove extra newline

**Pull request merged**